### PR TITLE
Default lm::link attribute to 'http://'.

### LINKED ISSUE #1013: learning material weblinks require "http://" prefix
if the user does not provide the `http://` prefix to a web address, adding the learning material weblink will fail. This is an onerous action to put on the user.

Solution: if possible, we should provide the `http://` as the default value on focus of the cursor in the address form field; if that seems too complex and hoop-jumpery, we should simply allow un-validated web addresses to be added.

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes #1013

selection_028

### CODE REVIEW

**Comment by saschaben** [comment]
:+1:

File: app/models/learning-material.js [hunk 1]
```diff
@@ -45,4 +45,13 @@ export default DS.Model.extend({
     }
     
   }),
+  isFile: computed('link', 'citation', 'absoluteFileUri', function() {
```
Comment (jrjohnson) [inline app/models/learning-material.js]: this can just depend on type, type will depend on the others.

Comment (stopfstedt) [inline app/models/learning-material.js]: tried that, didn't work. will try again.

**Comment by stopfstedt** [comment]
@jrjohnson defaulting the `link` attribute to `http://` works fine, however that data get's passed to the server on PUT/POST, polluting the `learning_materials` table. 

We can handle this either on the client or server-side, I'm leaning towards the latter. Thoughts?

**Comment by jrjohnson** [comment]
Ah right - so if its a citation you still end up with a link... yea.  How about we detect non fully qualified URLs on setLink on the entity and then add an http:// prefix there?  You could then return to checking for a URL as well.

**Comment by saschaben** [comment]
a little uncomfortable with adding the prefix value after the user submits --- what about `https://`or `ftp://` or `ftps://`or other ....? I'd rather let them just throw garbage at the server in some respects.

**Comment by jrjohnson** [comment]
I was thinking we would just detect any :// and not prefix those.

**Comment by stopfstedt** [comment]
found a better solution for populating newly instantiated weblink-LMs that leaves the link attribute blank for non-weblinks.

**Comment by jrjohnson** [comment]
@saschaben please review

**Comment by saschaben** [comment]
This is OK, I guess, as far as it goes. we will need to add validation (in another issue ticket) to disallow passing either a blank value or a value that = `http://` with no further trailing string as an address.

**Comment by saschaben** [comment]
:+1:

**Comment by jrjohnson** [comment]
@homu r+

**Pull request merged**